core/src/main/kotlin/fr/sncf/osrd/api/api_v2/pathfinding/PathfindingBlocksEndpointV2.kt

```diff
@@ -138,79 +147,192 @@ private fun computePaths(
-                                IncompatibleConstraints(
-                                    listOf(),
-                                    listOf(Pathfinding.Range(Offset.zero(), Offset.zero())),
-                                    listOf()
-                                )
-                            )
-                        )
-                    }
-                    SignalingSystemConstraints::class.java -> {
-                        throw NoPathFoundException(
-                            IncompatibleConstraintsPathResponse(
-                                relaxedPathResponse,
-                                IncompatibleConstraints(
-                                    listOf(),
-                                    listOf(),
-                                    listOf(
-                                        RangeValue(
-                                            Pathfinding.Range(Offset.zero(), Offset.zero()),
-                                            "signal"
-                                        )
-                                    )
-                                )
-                            )
-                        )
-                    }
-                }
-            }
+
+    if (
+        possiblePathWithoutErrorNoConstraints == null ||
+            possiblePathWithoutErrorNoConstraints.ranges.isEmpty()
+    ) {
+        return null
+    }
+
+    val pathRanges = possiblePathWithoutErrorNoConstraints.ranges
+
+    val elecConstraints = constraints.filterIsInstance<ElectrificationConstraints>()
+    assert(elecConstraints.size < 2)
+    val elecBlockedRangeValues =
+        getElecBlockedRangeValues(infra, pathRanges, elecConstraints.firstOrNull())
+
+    val gaugeConstraints = constraints.filterIsInstance<LoadingGaugeConstraints>()
+    assert(gaugeConstraints.size < 2)
+    val gaugeBlockedRanges =
+        getGaugeBlockedRanges(infra, pathRanges, gaugeConstraints.firstOrNull())
+
+    val signalingSystemConstraints = constraints.filterIsInstance<SignalingSystemConstraints>()
+    assert(signalingSystemConstraints.size < 2)
+    val signalingSystemBlockedRangeValues =
+        getSignalingSystemBlockedRangeValues(
+            infra,
+            pathRanges,
+            signalingSystemConstraints.firstOrNull()
+        )
+
+    if (
+        listOf(elecBlockedRangeValues, gaugeBlockedRanges, signalingSystemBlockedRangeValues).all {
+            it.isEmpty()
         }
+    ) {
+        return null
     }
-    // It didn’t fail due to a RS constraint, no path exists
-    throw NoPathFoundException(NotFoundInBlocks(listOf(), Length(0.meters)))
+
+    return IncompatibleConstraintsPathResponse(
+        runPathfindingPostProcessing(infra, possiblePathWithoutErrorNoConstraints),
+        IncompatibleConstraints(
+            elecBlockedRangeValues,
+            gaugeBlockedRanges,
+            signalingSystemBlockedRangeValues
+        )
+    )
+}
+
+private fun getElecBlockedRangeValues(
```

Comment:
You could probably do only one method here, 3 methods takes a lot of place not gonna lie (and it also means we'd have to change all 3 of em' if we need to remodify this code). Would only require to precompute the `pathProps` and pass the corresponding DistanceRangeMap as an argument (which also means we don't compute it 3 times, yay), and change the output: either keep it as is and do a condition for electrification, or just return a `DistanceRangeMap` and postprocess it (both are fine, but I feel like the first one would have less code and doesn't make the method too complex). What do you think?

Reply:
Thanks for this, I improved this in https://github.com/OpenRailAssociation/osrd/pull/7953/commits/65f3fe1e721d1210077d922190062d5ebb5a1d8a